sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkLogRecordBuilder.java

```diff
@@ -100,12 +103,18 @@ public <T> SdkLogRecordBuilder setAttribute(AttributeKey<T> key, T value) {
     if (key == null || key.getKey().isEmpty() || value == null) {
       return this;
     }
-    if (this.attributes == null) {
-      this.attributes =
-          AttributesMap.create(
-              logLimits.getMaxNumberOfAttributes(), logLimits.getMaxAttributeValueLength());
+    return setAttribute(key.asExtendedAttributeKey(), value);
```

Comment:
maybe we could cache the ExtendedAttributeKey inside of the AttributeKey to avoid the perf hit here

Reply:
> maybe we could cache the ExtendedAttributeKey inside of the AttributeKey to avoid the perf hit here

Already doing this! See changes to InternalAttributesKeyImpl